### PR TITLE
add unit tests for multiple IP aro create

### PR DESCRIPTION
**Pull request by tony-schndr**
### Which issue this PR addresses:
Adds unit tests for adding multiple managed IPs to the public load balancer during arm template generation

### What this PR does / why we need it:

Refactored public load balancer ARM template generation by isolating it into a separate function.  Added unit tests for the new function that tests the changes in #3110 

### Test plan for issue:

Unit tests
Manual test: 5 managed IPs, API server public
Manual test: 5 managed IPs, API Server private
Manual test: 1 managed IP, API Server private
E2E cluster creation: 1 managed IP, API Server public

### Is there any documentation that needs to be updated for this PR?
No

### CODE REVIEW

**Review by SudoBrendan — APPROVED** [review]

File: pkg/cluster/deploybaseresources.go [hunk 1]
```diff
@@ -201,6 +175,39 @@ func (m *manager) deployBaseResourceTemplate(ctx context.Context) error {
...
+	}
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs != nil {
+		for i := len(outboundIPs); i < m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.ManagedOutboundIPs.Count; i++ {
+			ipName := genManagedOutboundIPName()
+			*resources = append(*resources, m.networkPublicIPAddress(azureRegion, ipName))
+			outboundIPs = append(outboundIPs, api.ResourceReference{ID: m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID + "/providers/Microsoft.Network/publicIPAddresses/" + ipName})
+		}
+	}
+	m.patchEffectiveOutboundIPs(ctx, outboundIPs)
+	// Normal private clusters still need a public load balancer
```
Comment (kimorris27) [inline pkg/cluster/deploybaseresources.go:199]: I realize that this comment was just moved into a function and you didn't change it as part of this PR, but would you mind clarifying what it means either in this discussion or directly in the code (as appropriate)? It reads to me as if this code has been moved from a context where the comment made more sense.

File: pkg/cluster/deploybaseresources.go [hunk 2]
```diff
@@ -201,6 +175,39 @@ func (m *manager) deployBaseResourceTemplate(ctx context.Context) error {
...
+	*resources = append(*resources,
+		m.networkPublicLoadBalancer(azureRegion, outboundIPs),
+	)
+	// If the cluster is public we still want the default public IP address
+	if m.doc.OpenShiftCluster.Properties.IngressProfiles[0].Visibility == api.VisibilityPublic {
```
Comment (kimorris27) [inline pkg/cluster/deploybaseresources.go:204]: I didn't see a test case for this scenario (public ingress -> default PIP needed). If it makes sense to you, you could add one.

File: pkg/cluster/deploybaseresources.go (hunk 1 above)
Comment (tony-schndr) [reply pkg/cluster/deploybaseresources.go]: Agreed, I removed this comment.  I think it is clear in the code why we need a public load balancer (outbound routing).

File: pkg/cluster/deploybaseresources.go (hunk 2 above)
Comment (tony-schndr) [reply pkg/cluster/deploybaseresources.go]: Now that you point it out, it doesn't make sense to be in the `newPublicLoadBalancer` function.  It doesn't get added to the load balancer in the ARM template so I removed it from the function I created.

**Review by kimorris27 — APPROVED** [review]

**Review by SudoBrendan — APPROVED** [review]

**Review by cadenmarchese — APPROVED** [review]
lgtm!

**Pull request merged**